index.bs

```diff
@@ -16,13 +16,439 @@ WPT Display: closed
+ignore this header if it contains an invalid value.
+
+The following parameter is defined:
+
+* A parameter whose key is "`allowed-origin`", and whose value is a [=structured field/string=]. See below for processing requirements.
+
+<pre class="example" id="activate-storage-access-usage">
+
+// The server's response requests that the user agent activate storage access
+// before continuing with the load of the resource.
+Activate-Storage-Access: load
+
+// The server's response requests that the user agent activate storage access,
+// then retry the request. The "allowed-origin" parameter allowlists the
+// request's origin.
+Activate-Storage-Access: retry; allowed-origin="https://foo.bar"
+
+// Same as above, but using a wildcard instead of explicitly naming the request's Origin.
+Activate-Storage-Access: retry; allowed-origin=*
+</pre>
+
+<div algorithm>
+    To <dfn export>perform a storage access retry check</dfn> for a [=request=] |request| and [=response=] |response|, run the following steps:
+
+    <ol class="algorithm">
+        1. If |request|'s [=request/credentials mode=] [=string/is=] not "`include`", return failure.
+        1. If |request|'s [=request/eligible for storage-access=] is true, return failure.
+        1. Let |storageAccessStatus| be |request|'s [=request/storage access status=].
+        1. If |storageAccessStatus| [=string/is=] not "<code>[=storage access status/inactive=]</code>", return failure.
+        1. Let |parsedHeader| be the result of [=header list/get a structured field value|getting a structured field value=] given "`Activate-Storage-Access`" and "`item`" from |response|'s [=response/header list=].
+        1. If |parsedHeader| is null, return failure.
+        1. Let (|value|, |params|) be |parsedHeader|.
+        1. If |value| is not a Token, return failure.
+        1. If |value| [=string/is=] not "`retry`", return failure.
+        1. If |params|["allowed-origin"] does not exist, return failure.
+        1. Let |allowedOrigin| be |params|["allowed-origin"].
+        1. If |allowedOrigin| is a Token whose value is "`*`", return success.
+        1. If |allowedOrigin| is not a String, return failure.
+        1. Let |expectedOrigin| be the result of [=header list/get|getting=] "`Origin`" from |request|'s [=request/header list=].
+        1. If |expectedOrigin| is null, return failure.
+        1. If |allowedOrigin|'s value [=string/is=] |expectedOrigin|, return success.
+        1. Return failure.
+    </ol>
+</div>
+
+
+<div algorithm>
+    To <dfn export>perform a storage access load check</dfn> for a [=request=] |request| and [=response=] |response|, run the following steps:
+
+    <ol class="algorithm">
+        1. Let |storageAccessStatus| be |request|'s [=request/storage access status=].
+        1. If |storageAccessStatus| [=string/is=] not one of "<code>[=storage access status/inactive=]</code>" or "<code>[=storage access status/active=]</code>", return failure.
```

Comment:
I wonder if string/is is appropriate here, it's almost certainly an enum in implementations so we probably don't want to prescribe that.